port/unix/omrsignal.c

```diff
@@ -270,26 +295,17 @@ omrsig_can_protect(struct OMRPortLibrary *portLibrary,  uint32_t flags)
 	}
 #endif /* !defined(J9ZOS390) */
 
-
 	if (OMR_ARE_NO_BITS_SET(signalOptionsGlobal, OMRPORT_SIG_OPTIONS_REDUCED_SIGNALS_SYNCHRONOUS)) {
 		supportedFlags |= OMRPORT_SIG_FLAG_SIGALLSYNC;
 	}
 
-	if (OMR_ARE_NO_BITS_SET(signalOptionsGlobal, OMRPORT_SIG_OPTIONS_REDUCED_SIGNALS_ASYNCHRONOUS)) {
-		supportedFlags |= OMRPORT_SIG_FLAG_SIGQUIT | OMRPORT_SIG_FLAG_SIGABRT | OMRPORT_SIG_FLAG_SIGTERM;
-	}
-
-	if (OMR_ARE_ANY_BITS_SET(signalOptionsGlobal, OMRPORT_SIG_OPTIONS_SIGXFSZ)) {
-		supportedFlags |= OMRPORT_SIG_FLAG_SIGXFSZ;
-	}
-
 	if (OMR_ARE_ALL_BITS_SET(supportedFlags, flags)) {
 		Trc_PRT_signal_omrsig_can_protect_exiting_is_able_to_protect(supportedFlags);
 		return 1;
-	} else {
-		Trc_PRT_signal_omrsig_can_protect_exiting_is_not_able_to_protect(supportedFlags);
-		return 0;
 	}
+
+	Trc_PRT_signal_omrsig_can_protect_exiting_is_not_able_to_protect(supportedFlags);
```

Comment:
I think this tracepoint (and the one on line 303) should use `flags` rather than `supportedFlags `.

Reply:
no. these tracepoints expect `supportedFlags`:
```
TraceExit=Trc_PRT_signal_omrsig_can_protect_exiting_is_able_to_protect Group=signal Overhead=1 Level=2 NoEnv Template="omrsig_can_protect: Exiting and is able to protect, supportedFlags=0x%X"
TraceExit=Trc_PRT_signal_omrsig_can_protect_exiting_is_not_able_to_protect Group=signal Overhead=1 Level=2 NoEnv Template="omrsig_can_protect: Exiting and is not able to protect, supportedFlags=0x%X"
```
the entry tracepoint prints `flags`.

Reply:
Despite the wording of those messages, I think it makes more sense to say _which_ flags have been specified (and are supported) than emit the constant mask of _all_ supported flags.

Reply:
While debugging, one is expected to enable both entry and exit tracepoints. The entry tracepoint, `Trc_PRT_signal_omrsig_can_protect_entered`, prints the specified flags. The exit tracepoints, `Trc_PRT_signal_omrsig_can_protect_exiting_is_able_to_protect` and `Trc_PRT_signal_omrsig_can_protect_exiting_is_not_able_to_protect`, print the supported flags. The information from the entry and exit tracepoints can be used to determine if a specified flag is supported.

Reply:
Even if both entry and exit tracepoints are enabled, it does make the exit tracepoint less useful if it only ever says one thing.

Reply:
I feel the information provided by the below tracepoints is sufficient. Old tracepoints can't be amended. New tracepoints will need to be added to account for both `specified flags + supported flags`. It is not worth the effort to add more tracepoints to provide redundant information.

**Variant 1:**
```
omrsig_can_protect: Entered, flags=0x%x
omrsig_can_protect: Exiting and is able to protect, supportedFlags=0x%X
```
**Variant 2:**
```
omrsig_can_protect: Entered, flags=0x%x
omrsig_can_protect: Exiting and is not able to protect, supportedFlags=0x%X
```